### PR TITLE
add urls to logs

### PR DESCRIPTION
**Pull request by muhammad-ammar**
[EDUCATOR-1631](https://openedx.atlassian.net/browse/EDUCATOR-1631)

This will add `3play` and `cielo24` api urls into logs in case of an error. But before logging, sensitive information will be scrubbed from api url

### CODE REVIEW

**Comment by muhammad-ammar** [comment]
@muzaffaryousaf @mushtaqak @Qubad786 This is ready for review.

File: VEDA/utils.py [hunk 1]
```diff
@@ -69,3 +69,34 @@ def get_config(yaml_config_file=DEFAULT_CONFIG_FILE_NAME):
         config_dict = yaml.load(config)
 
     return config_dict
+
+
+def scrub_url_info(url, params_to_scrub):
```
Comment (muzaffaryousaf) [inline VEDA/utils.py]: maybe rename it to scrub_query_params ?

File: VEDA/utils.py [hunk 2]
```diff
@@ -69,3 +69,34 @@ def get_config(yaml_config_file=DEFAULT_CONFIG_FILE_NAME):
...
+    Returns:
+        url with query params scrubbed
+
+    >>> old_url = https://sandbox.veda.com/api/do?api_token=veda_api_key&job_name=12345&language=en&v=1
+    >>> new_url = https://sandbox.veda.com/api/do?v=1&job_name=12345&language=en&api_token=XXXXXXXXXXXX
+    """
+    parsed = urlparse.urlparse(url)
+
+    # query_params will be in the form of {'v': ['1'], 'job_name': ['12345'], 'api_token': ['cielo24_api_key']}
+    query_params = urlparse.parse_qs(parsed.query)
```
Comment (Qubad786) [inline VEDA/utils.py]: we can use `urlparse.parse_qsl` to avoid query parameters being single valued lists.

File: VEDA_OS01/transcripts.py [hunk 3]
```diff
@@ -19,7 +19,7 @@
 from rest_framework.views import APIView
```
Comment (Qubad786) [inline VEDA_OS01/transcripts.py]: missed out L#442

File: control/veda_deliver_3play.py [hunk 4]
```diff
@@ -105,11 +105,16 @@ def get_available_languages(self):
...
+        available_languages_url = build_url(self.base_url, self.available_languages_url, apikey=self.api_key)
+        response = requests.get(
+            url=available_languages_url
+        )
         if not response.ok:
             raise ThreePlayMediaLanguagesRetrievalError(
-                'Error while retrieving available languages: {response} -- {status}'.format(
-                    response=response.text, status=response.status_code
+                'Error while retrieving available languages: url={url} -- {response} -- {status}'.format(
+                    url=scrub_url_info(available_languages_url, ['apikey']),
```
Comment (Qubad786) [inline control/veda_deliver_3play.py]: missed out L#167

File: VEDA/utils.py [hunk 5]
```diff
@@ -69,3 +69,34 @@ def get_config(yaml_config_file=DEFAULT_CONFIG_FILE_NAME):
...
+        url (str): url
+        params_to_scrub (list): name of query params to be scrubbed from url
+
+    Returns:
+        url with query params scrubbed
+
+    >>> old_url = https://sandbox.veda.com/api/do?api_token=veda_api_key&job_name=12345&language=en&v=1
+    >>> new_url = https://sandbox.veda.com/api/do?v=1&job_name=12345&language=en&api_token=XXXXXXXXXXXX
+    """
+    parsed = urlparse.urlparse(url)
```
[the remark this answers is not in the document]
Comment (muhammad-ammar) [reply VEDA/utils.py]: but `urlparse.parse_qs` can unquote the query params itself and this is what we need.

File: control/veda_deliver_3play.py (hunk 4 above)
Comment (muhammad-ammar) [reply control/veda_deliver_3play.py]: Both the requests are post requests and we have payload available so we can directly log the interested params.

Comment (Qubad786) [reply control/veda_deliver_3play.py]: Ops right!

File: VEDA/utils.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (Qubad786) [reply VEDA/utils.py]: `unquote` will take care of that.

**Pull request merged**